File: src/main/scala/vm/vagrant/configuration/VagrantProviderConfig.scala
package vm.vagrant.configuration

/**
  * Created by oliver.ziegert on 24.03.17.
  */
/**
  * Configuration for a puppet provisioner that is used by Vagrant to configure a
  * VM
  *
  * @author oliver.ziegert
  *
  */
class VagrantProviderConfig(val name : String = "virtualbox",
                            val guiMode: Boolean = false,
                            val memory: Int,
                            val cpus: Int,
                            val customize: List[String],
                            val vmName: String)

File: src/main/java/vm/vagrant/configuration/VagrantProviderConfig.java
package vm.vagrant.configuration;

import java.util.List;

/**
 * Created by oliver.ziegert on 20.04.17.
 */
public class VagrantProviderConfig {
    private String name;
    private Boolean guiMode;
    private Integer memory;
    private Integer cpus;
    private List<String> customize;
    private String vmName;

    public VagrantProviderConfig(String name, Boolean guiMode, Integer memory, Integer cpus, List<String> customize, String vmName) {
        this.name = name;
        this.guiMode = guiMode;
        this.memory = memory;
        this.cpus = cpus;
        this.customize = customize;
        this.vmName = vmName;
    }

    public String name() {
        return name;
    }

    public Boolean guiMode() {
        return guiMode;
    }

    public Integer memory() {
        return memory;
    }

    public Integer cpus() {
        return cpus;
    }

    public List<String> customize() {
        return customize;
    }

    public String vmName() {
        return vmName;
    }
}
